src/topoflow_glacier/bmi/bmi_topoflow_glacier.py

```diff
@@ -272,242 +285,314 @@ def RH(self, value: np.ndarray) -> None:
-        self.Qn_LW = np.array([0], dtype="float64")
-        self.Q_sum = np.array([0], dtype="float64")
-        self.Qc = np.array([0], dtype="float64")
-        self.Qa = np.array([0], dtype="float64")
-        self.P_max = np.array([0], dtype="float64")
-        self.vol_P = np.array([0], dtype="float64")
-        self.vol_PR = np.array([0], dtype="float64")
-        self.vol_PS = np.array([0], dtype="float64")
-        self.Qn_SW = np.array([0], dtype="float64")
-        self.Qn_LW = np.array([0], dtype="float64")
-        self.Qn_tot = np.array([0], dtype="float64")
-        self.Q_sum = np.array([0], dtype="float64")
-        self.Qe = np.array([0], dtype="float64")
-        self.e_air = np.array([0], dtype="float64")
-        self.e_surf = np.array([0], dtype="float64")
-        self.em_air = np.array([0], dtype="float64")
-        self.Qc = np.array([0], dtype="float64")
-        self.Qa = np.array([0], dtype="float64")
-
-        # Ice component - constants can stay as scalars
-        self.rho_H2O = np.float64(self.cfg.rho_H2O)  # [kg/m**3]
-        self.rho_ice = np.float64(self.cfg.rho_ice)  # [kg/m**3]
-        self.Cp_ice = np.float64(self.cfg.Cp_ice)  # [J/(kg * K)]
-        self.Qg = np.float64(self.cfg.geothermal_heat_flux)  # [(J/yr)/m**2]
-        self.grad_Tz = np.float64(self.cfg.geothermal_gradient)  # [deg_C/m]
-        self.g = np.float64(self.cfg.g)  # [m/s**2]
-
-        # Glacier Component - constants can stay as scalars
-        self.rho_snow = np.float64(self.cfg.rho_snow)  # [kg/m**3]
-        self.Cp_snow = np.float64(self.cfg.Cp_snow)  # [J kg-1 K-1]
-        self.Lf = np.float64(self.cfg.Lf)  # [J kg-1]
-        self.T_rain_snow = np.float64(self.cfg.T_rain_snow)
 
-        # State variables - convert to 1D arrays
-        self.T0 = np.array([self.cfg.T0], dtype="float64")  # [deg C]
-        self.h_active_layer = np.array([self.cfg.h_active_layer], dtype="float64")  # [m]
-        self.mr_ice = np.array([0], dtype="float64")
-        self.vol_MR = np.array([0], dtype="float64")
-        self.meltrate = np.array([0], dtype="float64")
-
-        self._outputs.set_value(name="snowpack__depth", value=np.array([self.cfg.h0_snow], dtype="float64"))
-        self._outputs.set_value(
-            name="glacier_ice__thickness", value=np.array([self.cfg.h0_ice], dtype="float64")
-        )
-        self._outputs.set_value(
-            name="snowpack__liquid-equivalent_depth", value=np.array([self.cfg.h0_swe], dtype="float64")
-        )
-        self._outputs.set_value(
-            name="glacier__liquid_equivalent_depth", value=np.array([self.cfg.h0_iwe], dtype="float64")
-        )
+        # --- spatial constants ---
+        self.da_km2 = np.float64(self.cfg.da)
+        self.da_m2 = self.da_km2 * 1.0e6
+        self.slopes = np.array([self.cfg.slope], dtype="float64") if np.isscalar(self.cfg.slope) else np.asarray(self.cfg.slope, dtype="float64")
+
+        # --- timestep normalization: ensure dt is seconds ---
```

Comment:
It's your decision if you wanna run it on seconds. But I had changed  the code to hourly.